sled-agent/src/bootstrap/maghemite.rs

```diff
@@ -25,32 +25,33 @@ pub enum Error {
 
 pub async fn enable_mg_ddm_service(
     log: Logger,
-    interface: AddrObject,
+    interfaces: Vec<AddrObject>,
```

Comment:
We should probably either add an assertion or return an error if `interfaces` is empty, although I'm not sure which.

Reply:
It looks like we are checking this before we pass the `Vec<AddrObject>` to `enable_mg_ddm_service()`. `sled-agent` bails out if it doesn't discover any interfaces for `mg`.

https://github.com/oxidecomputer/omicron/blob/7eae7ea5747c00064048a3abfa03f1bd3b6cc876/sled-agent/src/bootstrap/server.rs#L73-L85

It also looks like I missed another comment I need to delete as well :man_facepalming:

Reply:
Ok perfect. I think I would still recommend adding an `assert!()` here then, both (a) to catch if this ever changes and (b) to make it clear to readers that callers should be checking this condition. It's probably worth a doc comment on this function to that effect:

```rust
/// Enable maghemite on the given interfaces.
///
/// # Panics
///
/// If `interfaces.is_empty()`.
```

Reply:
Hmm I see where you're coming from. However, since we're already returning `Result<(), Error>`, I'm leaning towards your earlier suggestion to return an `Error` if `interfaces` is empty. Panicking when we could just as easily return an `Error` seems like more work for everyone. :joy:

Reply:
That seems reasonable to me :+1: